app/src/main/java/com/framgia/oleo/screen/signup/SignUpFragment.kt

```diff
@@ -34,13 +41,67 @@ class SignUpFragment : DialogFragment(), Injectable {
         binding.viewModel = viewModel
         binding.setLifecycleOwner(this)
         val dialog = AlertDialog.Builder(activity)
-                .setView(binding.root)
-                .create()
+            .setView(binding.root)
+            .create()
         dialog.setCanceledOnTouchOutside(false)
         return dialog
     }
 
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setUpView()
+    }
+
+    private fun setUpView() {
+        buttonClose.setOnClickListener(this)
+        buttonSignUp.setOnClickListener(this)
+    }
+
+    override fun onClick(v: View?) {
+        when (v!!.id) {
+            R.id.buttonClose -> onCloseDialog()
+            R.id.buttonSignUp -> onSignUp()
+        }
+    }
+
+    private fun onSignUp() {
+        val email = editTextEmail.text.toString()
+        val userName = editTextUserName.text.toString()
+        val phoneNumber = editTextPhoneNumber.text.toString()
+        val password = editTextPassword.text.toString()
+        val progressDialog = ProgressDialog(context)
+        fireBase.createUserWithEmailAndPassword(email, password).addOnCompleteListener(activity!!) { task ->
+            if (task.isSuccessful) {
+                progressDialog.show()
+                val user = User()
+                user.userName = userName
+                user.email = email
+                user.phoneNumber = phoneNumber
```

Comment:
bỏ trực tiếp vào luôn cho gọn code nếu trường hợp không sử dụng thêm các biến  >1 lần e
```suggestion
                user.phoneNumber = editTextPhoneNumber.text.toString()
```